### PR TITLE
python3Packages.sjcl: init at 0.2.1

### PR DESCRIPTION
**Pull request by binsky08**
###### Description of changes

https://github.com/berlincode/sjcl

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1534,6 +1534,12 @@
     githubId = 185443;
     name = "Alexey Lebedeff";
   };
+  binsky = {
+    email = "timo@binsky.org";
+    github = "binsky08";
+    githubId = 30630233;
+    name = "Timo Triebensky";
+  };
```
Comment (mweinelt) [inline maintainers/maintainer-list.nix:1542]: Move this into a separate commit `maintainers: add binsky`, since you're touch a different file/attribute.

File: pkgs/development/python-modules/sjcl/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3761ba01f9f049eaa2f67516054b7bfb2e27fe8bc7c9c630da7168fba58f0c57";
+  };
+
+  propagatedBuildInputs = [ pycrypto ];
+
+  doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/sjcl/default.nix:22]: We enable tests for python packages since the ecosystem is quite fragile and we need to be able to notice breakages early or else.

The pypi sdist might not include them, but this should be verified. If that is the cae we might need to pull the source from GitHub using `fetchFromGitHub`

The check phase should look like so:

```nix
  checkPhase = ''
    runHook preCheck
    ${python.interpreter} -m unittest discover
    runHook postCheck
  '';
```

This requires `python` to be added to the function argument. Also note that we call `runHook` since the default `checkPhase` takes care of that, but we overwrite it, so we need that so `preCheck` and `postCheck` still work, if needed.

File: pkgs/development/python-modules/sjcl/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/sjcl/default.nix:23]:
```suggestion

  pythonImportsCheck = [
    "sjcl"
  ];
  
```

We also try to impor the modules that are documented, since it is another way to verify the package works.

File: pkgs/development/python-modules/sjcl/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+, fetchPypi
+, pythonOlder
+, pycrypto
+}:
+
+buildPythonPackage rec {
+  pname = "sjcl";
+  version = "0.2.1";
+
+  format = "pyproject";
```
Comment (mweinelt) [inline pkgs/development/python-modules/sjcl/default.nix:12]:
```suggestion
  format = "setuptools";
```

I can't find a `pyproject.toml` file in the repo or sdist.

File: pkgs/development/python-modules/sjcl/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3761ba01f9f049eaa2f67516054b7bfb2e27fe8bc7c9c630da7168fba58f0c57";
+  };
+
+  propagatedBuildInputs = [ pycrypto ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/sjcl/default.nix:20]: According to the sdist you're pulling it requires `pycryptodome`.

```
    install_requires=['pycryptodome'],
```

File: pkgs/development/python-modules/sjcl/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+, pythonOlder
+, pycrypto
+}:
+
+buildPythonPackage rec {
+  pname = "sjcl";
+  version = "0.2.1";
+
+  format = "pyproject";
+  disabled = pythonOlder "3.5";
```
Comment (mweinelt) [inline pkgs/development/python-modules/sjcl/default.nix:13]: The trove classifiers say 2.7, 3.5 and 3.6 are supported, so I think we might as well drop this statement entirely.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (binsky08) [reply maintainers/maintainer-list.nix:1554]: moved to commit [eca83bc](https://github.com/NixOS/nixpkgs/pull/167951/commits/eca83bc35d26795204a6b10593fbedadc7d256bc)

File: pkgs/development/python-modules/sjcl/default.nix (hunk 3 above)
Comment (binsky08) [reply pkgs/development/python-modules/sjcl/default.nix:22]: added with [e119b23](https://github.com/NixOS/nixpkgs/pull/167951/commits/e119b230877703a025cd50aa9b317f686ab46dd6)

File: pkgs/development/python-modules/sjcl/default.nix (hunk 6 above)
Comment (binsky08) [reply pkgs/development/python-modules/sjcl/default.nix]: removed with [e119b23](https://github.com/NixOS/nixpkgs/pull/167951/commits/e119b230877703a025cd50aa9b317f686ab46dd6)

File: pkgs/development/python-modules/sjcl/default.nix (hunk 4 above)
Comment (binsky08) [reply pkgs/development/python-modules/sjcl/default.nix]: changed with [e119b23](https://github.com/NixOS/nixpkgs/pull/167951/commits/e119b230877703a025cd50aa9b317f686ab46dd6)

File: pkgs/development/python-modules/sjcl/default.nix (hunk 2 above)
Comment (binsky08) [reply pkgs/development/python-modules/sjcl/default.nix]: added with 71564e2a8a06ac1b6840608b3753f806bb730c93

**Comment by dasJ** [comment]
How to properly switch base: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging
